In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('../08-Linear-Regression-Models/Advertising.csv')

In [3]:
df.head()

TV  radio  newspaper  sales
0  230.1   37.8       69.2   22.1
1   44.5   39.3       45.1   10.4
2   17.2   45.9       69.3    9.3
3  151.5   41.3       58.5   18.5
4  180.8   10.8       58.4   12.9

In [4]:
X = df.drop("sales", axis='columns')
y = df["sales"]

In [5]:
X.head()

TV  radio  newspaper
0  230.1   37.8       69.2
1   44.5   39.3       45.1
2   17.2   45.9       69.3
3  151.5   41.3       58.5
4  180.8   10.8       58.4

In [6]:
#Scaling the features
from sklearn.preprocessing import StandardScaler

In [7]:
scaler = StandardScaler()

In [8]:
from sklearn.model_selection import train_test_split

In [9]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=101)

In [10]:
scaler.fit(X_train)
X_train = scaler.transform(X_train)
X_test = scaler.transform(X_test)

In [11]:
#Rough random model
from sklearn.linear_model import Ridge

In [12]:
model = Ridge(alpha=100)

In [13]:
from sklearn.model_selection import cross_val_score

In [15]:
scores = cross_val_score(model, X_train, y_train, scoring='neg_mean_squared_error')

In [16]:
scores

array([ -9.32552967,  -4.9449624 , -11.39665242,  -7.0242106 ,
        -8.38562723])

In [17]:
abs(scores.mean())

8.215396464543606

As we can see from above is that the score is not too great, it needs to be improved upon

In [18]:
model_two = Ridge(alpha=1)

In [19]:
scores = cross_val_score(model_two, X_train, y_train, scoring='neg_mean_squared_error')

In [20]:
scores

array([-3.15513238, -1.58086982, -5.40455562, -2.21654481, -4.36709384])

In [21]:
abs(scores.mean())

3.344839296530695

As we can see, the score has improved quite a lot, by _updating_ our alpha value <br>
Now, we will use the test dataset, for our reporting purpose

In [22]:
model_two.fit(X_train, y_train)

Ridge(alpha=1)

In [23]:
test_predictions = model_two.predict(X_test)

In [24]:
from sklearn.metrics import mean_squared_error

In [25]:
MSE = mean_squared_error(y_test, test_predictions)

In [26]:
MSE

2.3190215794287514

2.32 is our final true error reporting metric, since it was tested on the data that it has never seen before